### PR TITLE
fix(generative-ai): Add tests for function_calling code snippet

### LINKED ISSUE #11018: Add missing test
Hi @koverholt  - it looks like this PR got merged without tests https://github.com/GoogleCloudPlatform/python-docs-samples/pull/11015 - please add a test following the guidelines in the authoring guide! https://github.com/GoogleCloudPlatform/python-docs-samples/blob/main/AUTHORING_GUIDE.md

cc @msampathkumar

### PR DESCRIPTION
**Pull request by koverholt**
## Description

Fixes #11018 by adding tests on function calling output.

## Checklist
- [x] I have followed [Sample Guidelines from AUTHORING_GUIDE.MD](https://github.com/GoogleCloudPlatform/python-docs-samples/blob/main/AUTHORING_GUIDE.md)
- [x] README is updated to include [all relevant information](https://github.com/GoogleCloudPlatform/python-docs-samples/blob/main/AUTHORING_GUIDE.md#readme-file)
- [x] **Tests** pass:   `nox -s py-3.9` (see [Test Environment Setup](https://github.com/GoogleCloudPlatform/python-docs-samples/blob/main/AUTHORING_GUIDE.md#test-environment-setup))
- [x] **Lint** pass:   `nox -s lint` (see [Test Environment Setup](https://github.com/GoogleCloudPlatform/python-docs-samples/blob/main/AUTHORING_GUIDE.md#test-environment-setup))
- [ ] These samples need a new **API enabled** in testing projects to pass (let us know which ones)
- [ ] These samples need a new/updated **env vars** in testing projects set to pass (let us know which ones)
- [ ] This sample adds a new sample directory, and I updated the [CODEOWNERS file](https://github.com/GoogleCloudPlatform/python-docs-samples/blob/main/.github/CODEOWNERS) with the codeowners for this sample
- [ ] This sample adds a new **Product API**, and I updated the [Blunderbuss issue/PR auto-assigner](https://github.com/GoogleCloudPlatform/python-docs-samples/blob/main/.github/blunderbuss.yml) with the codeowners for this sample
- [x] Please **merge** this PR for me once it is approved

### CODE REVIEW

File: generative_ai/function_calling.py [hunk 1]
```diff
@@ -0,0 +1,58 @@
...
+        prompt,
+        generation_config={"temperature": 0},
+        tools=[weather_tool],
+    )
+
+    return str(response)
+
+
+if __name__ == "__main__":
+    generate_function_call("What is the weather like in Boston?")
```
Comment (msampathkumar) [inline generative_ai/function_calling.py:56]: sugg: This can below the end region-tag or can be removed.

File: generative_ai/function_calling.py [hunk 2]
```diff
@@ -0,0 +1,58 @@
...
+    prompt = prompt
+
+    # Instruct the model to generate content using the Tool that you just created:
+    response = model.generate_content(
+        prompt,
+        generation_config={"temperature": 0},
+        tools=[weather_tool],
+    )
+
+    return str(response)
```
Comment (msampathkumar) [inline generative_ai/function_calling.py:52]:
```suggestion
    print(response)
    return response.text
```

File: generative_ai/function_calling.py (hunk 1 above)
Comment (koverholt) [reply generative_ai/function_calling.py:57]: I was wondering about this, thanks. Fixed in 5ea42e0 and moved outside of the region tag.

File: generative_ai/function_calling.py (hunk 2 above)
Comment (koverholt) [reply generative_ai/function_calling.py:53]: This is using the function calling feature that just landed in the latest `google-cloud-aiplatform` package earlier today, and for some reason when I ask for `response.text` or other variants/attributes that I've tried, I get:

```
In [7]: response.text
---------------------------------------------------------------------------
ValueError                                Traceback (most recent call last)
Cell In[7], line 1
----> 1 response.text

File ~/miniforge3/lib/python3.10/site-packages/vertexai/generative_models/_generative_models.py:1309, in GenerationResponse.text(self)
   1307 if len(self.candidates) > 1:
   1308     raise ValueError("Multiple candidates are not supported")
-> 1309 return self.candidates[0].content.parts[0].text

File ~/miniforge3/lib/python3.10/site-packages/vertexai/generative_models/_generative_models.py:1505, in Part.text(self)
   1502 @property
   1503 def text(self) -> str:
   1504     if "text" not in self._raw_part:
-> 1505         raise ValueError("Part has no text.")
   1506     return self._raw_part.text

ValueError: Part has no text.
```

Whereas getting the string representation of the response outputs the full contents that I can test against and users can work with.

**Comment by koverholt** [comment]
Tests are currently failing because this functionality just landed in `google-cloud-aiplatform==1.38.0` earlier today, whereas an older version of that client lib is being used in the tests. Also, tests will run against APIs that are not yet available but are scheduled to go live tomorrow.

cc @polong-lin

File: generative_ai/function_calling.py (hunk 2 above)
Comment (msampathkumar) [reply generative_ai/function_calling.py:53]: understood! Let me drop this.

**Comment by msampathkumar** [comment]
@koverholt CICD failing with below error. To fix, please update the required package in `requriements.txt` file

`ModuleNotFoundError: No module named 'vertexai.preview.generative_models`

**Comment by koverholt** [comment]
Thanks for the note! I updated the package version of `google-cloud-aiplatform` to 1.38.0, and now am hitting an (expected) permissions error for an allowlisted API that will be made public tomorrow. I can check back on this PR and re-run tests tomorrow.

**Comment by msampathkumar** [comment]
@koverholt Almost there! Check the error below

> Project `1012616486416` is not allowed to use Publisher Model `projects/python-docs-samples-tests/locations/us-central1/publishers/google/models/gemini-pro

**Review by kweinmeister — APPROVED** [review]

**Pull request merged**